bigbluebutton-html5/imports/ui/components/presentation/cursor/component.jsx

```diff
@@ -1,81 +1,289 @@
 
-const defaultProps = {
-  fill: 'red',
-  radius: 5,
-};
+    return obj;
+  }
 
-export default class Cursor extends Component {
-  constructor(props) {
-    super(props);
+  static getScaledSizes(props) {
+    const scaleFactor = props.widthRatio / 100 / props.physicalWidthRatio;
+
+    return {
+      // Adjust the radius of the cursor according to zoom
+      // and divide it by the physicalWidth ratio, so that svg scaling wasn't applied to the cursor
+      finalRadius: props.radius * scaleFactor,
+      // scaling the properties for cursorLabel and the border (rect) around it
+      cursorLabelText: {
+        textDY: props.cursorLabelText.textDY * scaleFactor,
+        textDX: props.cursorLabelText.textDX * scaleFactor,
+        fontSize: props.cursorLabelText.fontSize * scaleFactor,
+      },
+      cursorLabelBox: {
+        xOffset: props.cursorLabelBox.xOffset * scaleFactor,
+        yOffset: props.cursorLabelBox.yOffset * scaleFactor,
+        // making width and height a little bit larger than the size of the text
+        // received from BBox, so that the text didn't touch the border
+        width: (props.labelBoxWidth + 3) * scaleFactor,
+        height: (props.labelBoxHeight + 3) * scaleFactor,
+        strokeWidth: props.cursorLabelBox.labelBoxStrokeWidth * scaleFactor,
+      },
+    };
   }
 
-  render() {
+  componentWillMount() {
+    const cursorCoordinate = Cursor.getCursorCoordinates(this.props);
+    const { fill, displayLabel } = Cursor.getFillAndLabel(this.props);
+    const _scaledSizes = Cursor.getScaledSizes(this.props);
+
+    // setting the initial cursor info
+    this.cursorCoordinate = cursorCoordinate;
+    this.fill = fill;
+    this.displayLabel = displayLabel;
+    this.scaledSizes = _scaledSizes;
+  }
+
+  componentDidMount() {
+    // we need to find the BBox of the text, so that we could set a proper border box arount it
+    this.calculateCursorLabelBoxDimensions();
+  }
+
+  componentWillReceiveProps(nextProps) {
     const {
-      viewBoxWidth,
-      viewBoxHeight,
-      viewBoxX,
-      viewBoxY,
+      presenter,
+      isMultiUser,
       widthRatio,
+      physicalWidthRatio,
+      labelBoxWidth,
+      labelBoxHeight,
       cursorX,
       cursorY,
-      fill,
-      radius,
     } = this.props;
 
-    // Adjust the x,y cursor position according to zoom
-    const cx = (cursorX * viewBoxWidth) + viewBoxX;
-    const cy = (cursorY * viewBoxHeight) + viewBoxY;
+    if (presenter !== nextProps.presenter || isMultiUser !== nextProps.isMultiUser) {
+      const { fill, displayLabel } = Cursor.getFillAndLabel(nextProps);
+      this.displayLabel = displayLabel;
+      this.fill = fill;
+    }
+
+    if ((widthRatio !== nextProps.widthRatio ||
+          physicalWidthRatio !== nextProps.physicalWidthRatio)
+      ||
+        (labelBoxWidth !== nextProps.labelBoxWidth ||
+          labelBoxHeight !== nextProps.labelBoxHeight)) {
+      const _scaledSizes = Cursor.getScaledSizes(nextProps);
```

Comment:
same as above, you dont need to assign it to a variable before assign to the class propertie